src/docker-in-docker/install.sh

```diff
@@ -175,72 +177,67 @@ fi
+        echo $package_version_suffix
     fi
-fi
+}
 
 # Install Docker / Moby CLI if not already installed
 if type docker > /dev/null 2>&1 && type dockerd > /dev/null 2>&1; then
     echo "Docker / Moby CLI and Engine already installed."
 else
+    # Determine required packages and versions. Exit if they cannot be found.
+    engine_version_suffix=$(get_package_version_suffix "${engine_package_name}" "${DOCKER_VERSION}") || exit 1
+    cli_version_suffix=$(get_package_version_suffix "${cli_package_name}" "${DOCKER_VERSION}") || exit 1
+    required_packages="${cli_package_name}${cli_version_suffix} ${engine_package_name}${engine_version_suffix}"
+
+    # Moby always uses buildx
     if [ "${USE_MOBY}" = "true" ]; then
-        # Install engine
-        set +e # Handle error gracefully
-            apt-get -y install --no-install-recommends moby-cli${cli_version_suffix} moby-buildx${buildx_version_suffix} moby-engine${engine_version_suffix}
-            if [ $? -ne 0 ]; then
-                err "Packages for moby not available in OS ${ID} ${VERSION_CODENAME} (${architecture}). To resolve, either: (1) set feature option '\"moby\": false' , or (2) choose a compatible OS version (eg: 'ubuntu-20.04')."
-                exit 1
-            fi
-        set -e
+        moby_buildx_version_suffix=$(get_package_version_suffix "moby-buildx" "${MOBY_BUILDX_VERSION}") || exit 1
+        required_packages="${required_packages} moby-buildx${moby_buildx_version_suffix}"
+    fi
 
-        # Install compose
-        apt-get -y install --no-install-recommends moby-compose || err "Package moby-compose (Docker Compose v2) not available for OS ${ID} ${VERSION_CODENAME} (${architecture}). Skipping."
-    else
-        apt-get -y install --no-install-recommends docker-ce-cli${cli_version_suffix} docker-ce${engine_version_suffix}
-        # Install compose
-        apt-get -y install --no-install-recommends docker-compose-plugin || echo "(*) Package docker-compose-plugin (Docker Compose v2) not available for OS ${ID} ${VERSION_CODENAME} (${architecture}). Skipping."
+    # Install required packages (engine, CLI, and moby-buildx if moby)
+    set +e
+    apt-get -y install --no-install-recommends ${required_packages}
+    if [ $? -ne 0 ]; then
+        if [ "${USE_MOBY}" = "true" ]; then
+            err "Packages for moby not available in OS ${ID} ${VERSION_CODENAME} (${architecture}). To resolve, either: (1) set feature option '\"moby\": false' , or (2) choose a compatible OS version (eg: 'ubuntu-20.04')."
+        else
+            err "Packages for Docker not available in OS ${ID} ${VERSION_CODENAME} (${architecture}). To resolve, choose compatible OS and package versions in your devcontainer.json file."
+        fi
+        exit 1
     fi
+    set -e
+
+    # Install optional package, compose
+    compose_version_suffix=$(get_package_verson_suffix "${compose_package_name}" "${COMPOSE_PACKAGE_VERSION}")
```

Comment:
```suggestion
    compose_version_suffix=$(get_package_version_suffix "${compose_package_name}" "${COMPOSE_PACKAGE_VERSION}")
```

Typo: Cause for the test failures